### DISCUSSION TITLE
ML DAG examples in Argo

### DISCUSSION THREAD

**Discussion by saeid93** [argoproj/argo-workflows#7544] [Q&A] [answered]
I’m new to Argo workflow, I’m looking for some complete Argo only ML training/inference pipeline deployed on Argo. The only example I found in the internet is https://github.com/argoproj/argo-workflows/issues/929 which is using Kubeflow on top of Argo, but I am looking for some example which implement an ML pipeline only using only Argo. Examples on both inference and training will be very helpful. What I’m specifically trying to do is to replicate inference pipelines in this papers InferLine: [ML Prediction Pipeline Provisioning and Management for Tight Latency Objectives](https://arxiv.org/pdf/1812.01776.pdf) with Argo workflow. This is a Github repo of the [example code](https://github.com/hsubbaraj/osdi_pipelines) from the paper. Please share with me any similar Argo project that might give me ideas in the implementation of such inference pipelines. Training examples are also useful for me
Thank you in advance!

**Comment by terrytangyuan** [comment, marked as answer]
Currently, we only have one example workflow that includes a single TensorFlow model training step using Kubeflow training operator: https://github.com/argoproj/argo-workflows/blob/master/examples/k8s-resource-log-selector.yaml

In addition, I am working on this [Distributed Machine Learning Patterns](https://github.com/terrytangyuan/distributed-ml-patterns) book (still in progress) that will be using Argo Workflows to build an end-to-end ML system, including both training and serving that you mentioned. Stay tuned!

**Reply by saeid93** [reply]
Thank you for your answer! Looking forward to read your book!
